File: 02-two_pointers/00167-Two_Sum_II/00167-two_sum_II.cpp
// 167: Two Sum II
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    // SOLUTION
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;
        
        while (l<r) {
            int s = numbers[l] + numbers[r];
            if (s==target) return {l+1, r+1};
            else if (s<target) l++;
            else r--;
        }

        return {};
    }
};


int main() {
    Solution o;

    // INPUT
    vector<int> numbers = {2,7,11,15};
    int target = 9;

    // OUTPUT
    auto result = o.twoSum(numbers, target);
    cout<<"["; for (auto v : result) cout<<v<<" "; cout<<"\b]"<<endl;
}

File: 02-two_pointers/00167-Two_Sum_II/00167-two_sum_II.py
# 167: Two Sum II
# https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/


class Solution:
    # SOLUTION
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        l, r = 0, len(numbers) - 1

        while l<r:
            s = numbers[l] + numbers[r]
            if s==target: return [l+1, r+1]
            elif s<target: l+=1
            else: r-=1
        
        return []


if __name__ == "__main__":
    o = Solution()

    # INPUT
    numbers = [2,7,11,15]
    target = 9

    # OUTPUT
    result = o.twoSum(numbers, target)
    print(result)